src/doc/tutorials/PythonExpressions.rst

```diff
@@ -0,0 +1,202 @@
+Changes to the script can be made either by modifying the script in the *Python expression script* editor or modifying it in an external text editor and then reloading the script.
+Generaly speaking, modifying the script in the *Python expression script* editor is easier than doing it in an external text editor except that it is difficult to tell how many spaces are at the beginning of the line since the editor doesn't use a fixed width font.
+
+Developing the script in the *Python expression script* editor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following steps can be used to iteratively develop your script using the *Python expression script* editor.
+
+1. Edit the script.
+2. Click the *Apply* button.
+3. Go to *Variables->d* to change the variable to *d*.
+   The first time you try your script this won't be necessary since the variable is already *d*.
+4. Go to *Variables->MyExpression* to change the variable to "MyExpression" and execute the script.
+
+Developing the script in a text editor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
+
+The following steps can be used to iteratively develop your script using a text editor.
+
+Before you can modify the script you will need to save it.
+
+1. Click *Save script* to save the script.
+
+Now you are ready to modify the script.
+
+2. Edit the script with your favorit editor.
+3. Go to *Load script->File* to load the script.
+4. Click the *Apply* button.
+5. Go to *Variables->d* to change the variable to *d*.
+   The first time you try your script this won't be necessary since the variable is already *d*.
+6. Go to *Variables->MyExpression* to change the variable to "MyExpression" and execute the script.
+ 
+Python Expression Example 1
+---------------------------
+
+This example will operate on 2D meshes and takes the distance around the edges of each cell and multiplies it by the value of the cell.
+It demonstrates accessing the coordinates and topology of the mesh as well as a variable.
+
+Here is the example script.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 11-48
+
+Let's start off by creating a Pseudocolor plot from the expression.
+
+1. Copy the script into the *Python expression script* editor.
+2. Click the *Apply* button.
+3. Go to *Variables->MyExpression* to change the variable to the expression.
+
+.. figure:: images/PythonExpression-Plot1.png
+
+   The Pseudocolor plot of MyExpression
+
+Now let's take a look at the script and see what each portion does.
+
+The ``__init__`` method provides information about the expression, including
+
+* That it inherits from ``SimplePythonExpression``.
+* The name of the expression.
+* A description of the expression.
+* A flag indicating that the output is not a point centered value.
+* A flag that the output is a scalar.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 13-18
```

Comment:
I worry about using the `:lines:` directive for this. I mean, it works...now. But if anyone changes the script in any way *above* these lines, it silently corrupts what gets rendered in our docs.

I think using regex is better. You can put `# key A {` and `# key A }` (or some such convention) at end of lines you want to bracket or add a sep. comment lines.

I know this makes the python code look a little less like it would ordinarily. I am not sure there is a better option.

Reply:
I think adding some type of tags will make the script really hard to read when looking at the script in it's entirety. I added a very prominent warning at the top of each of the two scripts that are referenced in the tutorial that the line numbers need to be updated in the Python Expressions tutorial if the scripts are changed.